### DISCUSSION TITLE
i have PostgreSQL as the database, but how can I have fuzzy search for my app?

### DISCUSSION THREAD

**Discussion by tesla-cat** [fastapi/fastapi#8367] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
na
```

### Description

- I am a beginner and I want to make an app
- I learnt that I can use PostgreSQL and sqlalchemy to handle the database part
- I tried the PostgreSQL Full Text Search functionality but the performance is bad
- I wonder how can I have fuzzy search functionality using FastAPI?

### Operating System

Windows

### Operating System Details

na

### FastAPI Version

na

### Python Version

na

**Comment by JarroVGIT** [comment, marked as answer]
I am afraid that this is not a FastAPI related question. Fuzzy search of text, in combination with a database, is a database related question. If you are using Postgres and the performance is bad, try another database such as [sphinx](https://sphinxsearch.com/docs/sphinx3.html#getting-started-on-linux-and-macos) . But as far as FastAPI goes, this project has not much to do with that part of your solution.

**Comment by imacat** [comment]
I suppose this is not a PostgreSQL/SQLAlchemy question, not FastAPI question.

And, as far as I know, there is no fuzzy search to PostgreSQL.  The fuzzy search is a unique feature of MySQL.  If you do need fuzzy search, your only chance is to turn to MySQL.
